tree/treeplayer/inc/ROOT/TVec.hxx

```diff
@@ -0,0 +1,535 @@
+          typename D = typename std::enable_if<!ROOT::Detail::VecOps::TIsTVec<V>::value, int>::type>
+TVec<Boolean_t> operator==(const TVec<T> &v, const V &c)
+{
+   return ROOT::Internal::VecOps::Operate(v, [&c](const T &t) -> Boolean_t { return t == c; });
+}
+
+template <typename T, typename V,
+          typename D = typename std::enable_if<!ROOT::Detail::VecOps::TIsTVec<V>::value, int>::type>
+TVec<Boolean_t> operator!=(const TVec<T> &v, const V &c)
+{
+   return ROOT::Internal::VecOps::Operate(v, [&c](const T &t) -> Boolean_t { return t != c; });
+}
+
+template <typename T, typename V,
+          typename D = typename std::enable_if<!ROOT::Detail::VecOps::TIsTVec<V>::value, int>::type>
+TVec<Boolean_t> operator<=(const TVec<T> &v, const V &c)
+{
+   return ROOT::Internal::VecOps::Operate(v, [&c](const T &t) -> Boolean_t { return t <= c; });
+}
+
+template <typename T, typename V,
+          typename D = typename std::enable_if<!ROOT::Detail::VecOps::TIsTVec<V>::value, int>::type>
+TVec<Boolean_t> operator<(const TVec<T> &v, const V &c)
+{
+   return ROOT::Internal::VecOps::Operate(v, [&c](const T &t) -> Boolean_t { return t < c; });
+}
+///@}
+
+/** @name Math Operators with TVecs
+ *  Math operators involving TVecs
+*/
+///@{
+
+template <typename T, typename V>
+auto operator+(const TVec<T> &v0, const TVec<V> &v1) -> TVec<decltype(v0[0] + v1[0])>
+{
+   return ROOT::Internal::VecOps::Operate(v0, v1, "+", [](const T &t, const V &v) { return t + v; });
+}
+
+template <typename T, typename V>
+auto operator-(const TVec<T> &v0, const TVec<V> &v1) -> TVec<decltype(v0[0] - v1[0])>
+{
+   return ROOT::Internal::VecOps::Operate(v0, v1, "-", [](const T &t, const V &v) { return t - v; });
+}
+
+template <typename T, typename V>
+auto operator*(const TVec<T> &v0, const TVec<V> &v1) -> TVec<decltype(v0[0] * v1[0])>
+{
+   return ROOT::Internal::VecOps::Operate(v0, v1, "*", [](const T &t, const V &v) { return t * v; });
+}
+
+template <typename T, typename V>
+auto operator/(const TVec<T> &v0, const TVec<V> &v1) -> TVec<decltype(v0[0] / v1[0])>
+{
+   return ROOT::Internal::VecOps::Operate(v0, v1, "/", [](const T &t, const V &v) { return t / v; });
+}
+
+template <typename T, typename V>
+auto operator%(const TVec<T> &v0, const TVec<V> &v1) -> TVec<decltype(v0[0] % v1[0])>
+{
+   return ROOT::Internal::VecOps::Operate(v0, v1, "%", [](const T &t, const V &v) { return t % v; });
+}
+
+template <typename T, typename V>
+TVec<Boolean_t> operator>(const TVec<T> &v0, const TVec<V> &v1)
+{
+   return ROOT::Internal::VecOps::Operate(v0, v1, ">", [](const T &t, const V &v) -> Boolean_t { return t > v; });
+}
+
+template <typename T, typename V>
+TVec<Boolean_t> operator>=(const TVec<T> &v0, const TVec<V> &v1)
+{
+   return ROOT::Internal::VecOps::Operate(v0, v1, ">=", [](const T &t, const V &v) -> Boolean_t { return t >= v; });
+}
+
+template <typename T, typename V>
+TVec<Boolean_t> operator==(const TVec<T> &v0, const TVec<V> &v1)
```

Comment:
p.s. I understand that it's desirable for `operator==` to behave like this, I was just thinking out loud here...

Reply:
you have a point. Here I do not know really how to proceed. I think that in case of TVec  and therewith analysis `operator==` should really behave as it is doing now. TVec is born to help analysis. If a "proxying vector" is needed, one can directly use vector<T, TAdoptAllocator<T>>, right?